scripts/install_backend_python_libs.py

```diff
@@ -88,12 +113,20 @@ def _remove_metadata(library_name, version_string):
         version_string: str. Stringified version of the library to remove the
             metadata for.
     """
-    possible_file_names = _get_possible_metadata_directory_names(
+    possible_directory_names = _get_possible_metadata_directory_names(
         library_name, version_string)
-    for f in os.listdir(common.THIRD_PARTY_PYTHON_LIBS_DIR):
-        if f.lower() in possible_file_names:
+    for file_or_directory in os.listdir(common.THIRD_PARTY_PYTHON_LIBS_DIR):
+        # Python metadata directory names contain a python library name that
+        # does not have uniform case. However, python libraries are equivalent
+        # regardless of their case. Therefore, in order to check if a python
+        # library's metadata exists in a directory, we need to normalize the
+        # directory name. Otherwise, we would need to check every permutation of
+        # the casing for metadata directories generated with the naming
+        # convention: <library_name>-<library-version>.
+        if (_normalize_string_that_contains_python_library_name(
```

Comment:
Would something like this help, so that we can deal with only dirs? https://stackoverflow.com/questions/141291/how-to-list-only-top-level-directories-in-python

Reply:
I chose to use this solution
```
[d for d in os.listdir(folder) if os.path.isdir(os.path.join(folder, d))]
```
because it is the clearest in what it supposed to do and it is actually faster than the other solution (next(os.walk)).